sympy/assumptions/assume.py

```diff
@@ -156,10 +156,10 @@ def __new__(cls, name, handlers=None):
         return obj
 
     def _hashable_content(self):
-        return (self.name,)
+        return (self.name, tuple(self.handlers))
```

Comment:
This is possibly dangerous - it means that the hash changes when handlers are added and removed